### ISSUE TITLE
WCAG 1.3.1: Ensures elements with an ARIA role that require child roles contain them (#DatePicker50-label).

### ISSUE THREAD

**Issue by Payne-Yan** [microsoft/fluentui#14461]
**Environment Information**

- **Package version(s)**: 7.115.1
- **Browser and OS versions**: Chrome Version 84.0.4147.105 (Official Build) (64-bit)

**Describe the issue:**

Ensures elements with an ARIA role that require child roles contain them

**Element path**
#DatePicker178-label

Snippet

**How to fix** 
  Fix the following
  Required ARIA child role not present: listbox

**Actual behavior:**

elements with an ARIA role that requires child roles do not contain them (#DatePicker50-label).

**Expected behavior:**

ensures elements with an ARIA role that require child roles contain them (#DatePicker50-label).

**Documentation describing expected behavior**

**Comment by Payne-Yan** [comment]
Repro-Image
![image](https://user-images.githubusercontent.com/69501015/89875503-e8f10e00-dbef-11ea-8d9c-8141984017f7.png)

**Comment by paulgildea** [comment]
@smhigley FYI

**Comment by khmakoto** [comment]
Hi @Payne-Yan, I've tried reproducing this in our website unsuccessfully but I see you are a few versions behind. Could you try with our latest version and reply back? Thanks!

![image](https://user-images.githubusercontent.com/7798177/90077239-857de380-dcb6-11ea-98fa-52fda011a24b.png)

**Comment by Payne-Yan** [comment]
@khmakoto Repro steps
1. Navigate to date picker combobox and select the picker, calendar pops up will appear, navigate on the pop, and select date.
2. Open accessibility insights for web tools and select fast pass options.
3. Verify, whether the elements with an ARIA role that require child roles contain them(#DatePicker50-label) or not.

**Comment by khmakoto** [comment]
@Payne-Yan I've gone through the process once again and I just can't seem to repro this behavior as much as I try. Hence I'll be resolving this as `Resolution: Can't repro`. Thanks and let us know if there is anything else I can do to help!

![image](https://user-images.githubusercontent.com/7798177/90287377-99e3ec80-de2c-11ea-8eb2-c7a622448271.png)

**Comment by Payne-Yan** [comment]
@khmakoto According to my steps to reproduce, it can be reproduced.
![image](https://user-images.githubusercontent.com/69501015/90349672-3c8aae00-e06d-11ea-8bea-6dc9896f8cee.png)

**Comment by Payne-Yan** [comment]
@khmakoto More importantly, it needs to pop up the calendar and then open accessibility insights for the web tools to reproduce this scene.

**Comment by smhigley** [comment]
@Payne-Yan it looks like this is because axe-core hasn't updated to the ARIA 1.2 combobox pattern, where the combobox role is on the input and associated with the listbox/grid/etc through `aria-controls`. You can see the same failure if you run Accessibility Insights on the aria-practices combobox page: https://w3c.github.io/aria-practices/examples/combobox/combobox-autocomplete-list.html

In practice, this combobox pattern has much better screen reader support than the 1.1 pattern, so this should be a bug on axe-core and not on Fluent.

**Comment by khmakoto** [comment]
Thanks for responding here @smhigley!

@Payne-Yan given @smhigley's response I'm going to go ahead and resolve this as `Resolution: External`. Let us know if there is anything else we can do to help!

**Comment by Payne-Yan** [comment]
@smhigley I submitted the axe-core GitHub issue and waited for their reply. Is it an axe-core problem?

**Comment by khmakoto** [comment]
Yes, that seems to be the case.

**Comment by smhigley** [comment]
Yes, though it's slightly complicated by ARIA 1.2 still being a Working Draft and not a full recommendation. I still think axe-core should remove that required children combobox rule, though. The justifications would be:
- Even though the 1.2 spec is not a recommendation yet, the new combobox pattern is pretty set in stone at this point
- The browser/screen reader support for the 1.2 pattern is already much better than the 1.1 pattern, and practical accessibility should trump purity for checkers like axe
- Based on the above, I'd say it's currently a false positive

I'd guess the working draft vs. recommendation thing is the reason they haven't removed it yet. I hadn't seen any issues on that before yours though, so I don't know what their stance on it is.

**Comment by Payne-Yan** [comment]
@smhigley Their answer:
![image](https://user-images.githubusercontent.com/69501015/91130873-f1ad0e00-e6de-11ea-9b71-902ea36b99c3.png)

axe-core issue: https://github.com/dequelabs/axe-core/issues/2478

**Comment by WilcoFiers** [comment]
> The browser/screen reader support for the 1.2 pattern is already much better than the 1.1 pattern, and practical accessibility should trump purity for checkers like axe

@smhigley Do you have any data to show support for either pattern? If you're right about this, that's definitely something we should change in axe-core. 

I do think there are some issues with how the data picker is implemented. Users need `aria-haspopup="modal"` so that screen readers can provide the correct instructions on how to interact with it, and aria-controls should be assigned to the actual modal, instead of to its parent.

**Comment by smhigley** [comment]
@WilcoFiers If you don't mind, I'm going to answer this over in the axe-core thread, since it's not specific to Fluent :)

**Comment by khmakoto** [comment]
Thanks for all your support @smhigley!

**Comment by Payne-Yan** [comment]
Could you please solve this issue on the official website first? We can make a reference.

**Comment by Payne-Yan** [comment]
@khmakoto Could you please solve this issue on the official website first?

**Comment by Payne-Yan** [comment]
@khmakoto The date picker on the https://developer.microsoft.com/en-us/fluentui#/controls/web/datepicker is solved first.

**Comment by khmakoto** [comment]
@Payne-Yan, like explained by @smhigley, this happens because axe-core does not support the 1.2 standard yet. I don't think there's an actionable item on our side regarding this issue.

**Comment by Payne-Yan** [comment]
@khmakoto On the official website, click Accessibility Insights for Web tool, the pop-up box of date picker is closed directly, there is no way to reproduce it. Did you update again?
